### PR TITLE
Rework deb packaging and integrate wetty

### PR DESCRIPTION
**Pull request by peppsac**
We don't add node(js) and npm as dependencies to not break user setup.
Wetty is packaged with ronin but in the long term we probably want
to have 2 separate packages.

Documentation references for the packaging:
https://web.archive.org/web/20141210051522/http://developer.ubuntu.com/2012/02/how-to-prepare-a-compiled-application-for-ubuntu-software-center/
https://wiki.ubuntu.com/SystemdForUpstartUsers
https://wiki.debian.org/IntroDebianPackaging

@autra r? (this depends on https://github.com/Phoxygen/wetty/pull/1)

[Review on Reviewable](https://reviewable.io/reviews/phoxygen/b2gian/6)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -88,7 +91,14 @@ gaia-sources: ${GAIA_SOURCE_DIR}
 		exit 1; \
 	fi
 
-sources: gecko-sources gaia-sources
+${WETTY_SOURCE_DIR}:
+	git clone https://github.com/Phoxygen/wetty
```
Comment (autra) [inline Makefile]: `git clone https://github.com/Phoxygen/wetty ${WETTY_SOURCE_DIR}` ?

File: Makefile [hunk 2]
```diff
@@ -138,28 +148,53 @@ run: ${BUILD_DIR}/.runtime_deps_ready ${BUILD_DIR}/gecko/dist/b2g ${B2G_PROFILE_
...
+	tar --directory ${PACKAGE_DIR} -xjf ${BUILD_DIR}/gecko/dist/b2g-${GECKO_VERSION}.0a1.en-US.linux-x86_64.tar.bz2
+	cp ${BUILD_DIR}/gaia/profile.tar.bz2 ${PACKAGE_DIR} 
+	cp ${mkfile_dir}/launch.sh ${PACKAGE_DIR}
+	cp ${mkfile_dir}/session.sh ${PACKAGE_DIR}
+	cp ${mkfile_dir}/ronin.desktop ${PACKAGE_DIR}
+
+	cd ${WETTY_SOURCE_DIR} && cp -r app.js package.json public ${PACKAGE_DIR}/wetty
+
+	# Generate ${PACKAGE_DIR}/debian folder
+	cd ${PACKAGE_DIR} && dh_make -n -s -y -e tech@phoxygen.com
```
Comment (autra) [inline Makefile]: I always had the feeling `dh_make` should be run once, then that the debian folders should be committed and used by other dh-script, WDYT?

File: Makefile [hunk 3]
```diff
@@ -138,28 +148,53 @@ run: ${BUILD_DIR}/.runtime_deps_ready ${BUILD_DIR}/gecko/dist/b2g ${B2G_PROFILE_
...
+		cp wetty.conf ${PACKAGE_DIR}/debian/ronin.upstart
+
+	# Fixup generated debian folder
+	cp ${mkfile_dir}/debian/install ${PACKAGE_DIR}/debian
+	cp ${mkfile_dir}/debian/postinst ${PACKAGE_DIR}/debian
+	cp ${mkfile_dir}/debian/prerm ${PACKAGE_DIR}/debian
+	sed -i '/^Build-Depends:/ s/$$/, dh-systemd/' ${PACKAGE_DIR}/debian/control
+	# sed -i '/^Depends:/ s/$$/, nodejs, npm/' ${PACKAGE_DIR}/debian/control
+	head -n -2  ${PACKAGE_DIR}/debian/control | tee ${PACKAGE_DIR}/debian/control
+	sed -i '/^\tdh/ s/$$/--with=systemd/' ${PACKAGE_DIR}/debian/rules
```
Comment (autra) [inline Makefile]: Committing the result of dh_make would maybe help us avoid these runtime file manipulation, WDYT?

File: Makefile (hunk 2 above)
Comment (autra) [inline Makefile]: Ok forget this, I didn't see the files below.

Comment (autra) [inline Makefile]: Actually, you've committed them, so why do you need to execute again dh_make?

File: Makefile [hunk 4]
```diff
@@ -138,28 +148,53 @@ run: ${BUILD_DIR}/.runtime_deps_ready ${BUILD_DIR}/gecko/dist/b2g ${B2G_PROFILE_
...
+	cd ${WETTY_SOURCE_DIR}/upstart && \
+		cp wetty.conf ${PACKAGE_DIR}/debian/ronin.upstart
+
+	# Fixup generated debian folder
+	cp ${mkfile_dir}/debian/install ${PACKAGE_DIR}/debian
+	cp ${mkfile_dir}/debian/postinst ${PACKAGE_DIR}/debian
+	cp ${mkfile_dir}/debian/prerm ${PACKAGE_DIR}/debian
+	sed -i '/^Build-Depends:/ s/$$/, dh-systemd/' ${PACKAGE_DIR}/debian/control
+	# sed -i '/^Depends:/ s/$$/, nodejs, npm/' ${PACKAGE_DIR}/debian/control
+	head -n -2  ${PACKAGE_DIR}/debian/control | tee ${PACKAGE_DIR}/debian/control
```
Comment (autra) [inline Makefile]: This line removes everything from `${PACKAGE_DIR}/debian/control except the first 2 lines right? Why so?

File: Makefile (hunk 3 above)
Comment (autra) [inline Makefile]: mhh actually not sure I understand this line. Could you explain to me?

File: debian/install [hunk 5]
```diff
@@ -0,0 +1,6 @@
+b2g/* /opt/b2g/b2g/
```
Comment (autra) [inline debian/install:1]: Do we really need that list of `b2g`? that being said, that's not the point of this PR, so we could refactor that later.

File: Makefile (hunk 4 above)
Comment (peppsac) [inline Makefile]: Nope, it keeps everything but the last 2

File: Makefile (hunk 3 above)
Comment (peppsac) [inline Makefile]: sed magic :-)

Comment (peppsac) [inline Makefile]: Replaces:
`dh $@`
By:
`dh $@ --with=systemd`

File: Makefile (hunk 1 above)
Comment (peppsac) [inline Makefile]: Good catch

File: Makefile (hunk 2 above)
Comment (peppsac) [inline Makefile]: Yeah, I will redo this and commit the result `dh_make -n` and remove the patching lines

**Comment by peppsac** [comment]
@autra r?

**Comment by autra** [comment]
So I have symlinked /usr/bin/node, and now I have this error when installing ronin

```
gyp: /home/augustin/.node-gyp/4.2.4/common.gypi not found (cwd: /opt/b2g/wetty/node_modules/pty.js) while reading includes of binding.gyp while trying to load binding.gyp
dpkg: error processing package ronin (--install):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for ureadahead (0.100.0-19) ...
Errors were encountered while processing:
 ronin
```

do you know what this is about? I've already tried to remove my `~/.node-gyp` folder without any luck.

**Comment by autra** [comment]
LGTM, merging ! Sorry for the delay ;-)

**Pull request merged**